src/engine/client/cl_cgame.cpp

```diff
@@ -1102,6 +1102,32 @@ void CGameVM::QVMSyscall(int syscallNum, Util::Reader& reader, IPC::Channel& cha
 			});
 			break;
 
+		case CG_CM_MARKFRAGMENTSARRAY:
+			IPC::HandleMsg<CMMarkFragmentsArrayMsg>(channel, std::move(reader), [this] (
+				const std::vector<fragmentPacket_t>& fragmentPacketsArray,
+				std::vector<fragmentBuffer_t>& fragmentBuffersArray)
+			{
+				size_t numMarks = fragmentPacketsArray.size();
+
+				fragmentBuffersArray.resize(numMarks);
+
+				for ( size_t m = 0; m < numMarks; m++ )
+				{
+					const fragmentPacket_t *fragmentPacket = &fragmentPacketsArray[ m ];
+					fragmentBuffer_t *fragmentBuffer = &fragmentBuffersArray[ m ];
+
+					fragmentBuffer->numFragments = re.MarkFragments(
+						4,
+						(const vec3_t*) fragmentPacket->originalPoints.data(),
+						fragmentPacket->projection.data(),
+						MAX_MARK_POINTS,
+						fragmentBuffer->markPoints.data()[ 0 ].data(),
```

Comment:
`data()[ 0 ]` is just `[ 0 ]`?

Reply:
```
src/engine/client/cl_cgame.cpp:1124:80: error: invalid types ‘<unresolved overloaded function type>[int]’ for array subscript
 1124 |                                                 fragmentBuffer->markPoints.data[ 0 ].data(),
      |                                                                                ^
```

Reply:
From to-be-squashed commits in:

- https://github.com/Unvanquished/Unvanquished/pull/2981

Here are some of the micro-rewriting-steps leading to this code:

```diff
diff --git a/src/cgame/cg_marks.cpp b/src/cgame/cg_marks.cpp
index 507fabc63..dba1eaef0 100644
--- a/src/cgame/cg_marks.cpp
+++ b/src/cgame/cg_marks.cpp
@@ -183,7 +183,7 @@ void CG_ProcessImpactMarks()
 		byte           colors[ 4 ];
 		int            numFragments;
 		std::array<markFragment_t, MAX_MARK_FRAGMENTS> markFragments;
-		vec3_t         markPoints[ MAX_MARK_POINTS ];
+		std::array<vec3_t, MAX_MARK_POINTS> markPoints;
 		vec3_t         projection;
 
 		//if ( markTotal >= MAX_MARK_POLYS ) {
@@ -210,7 +210,7 @@ void CG_ProcessImpactMarks()
 		// get the fragments
 		VectorScale( dir, -20, projection );
 		numFragments = trap_CM_MarkFragments( 4, ( const vec3_t * ) originalPoints,
-			projection, MAX_MARK_POINTS, markPoints[ 0 ],
+			projection, MAX_MARK_POINTS, markPoints.data()[ 0 ],
 			MAX_MARK_FRAGMENTS, markFragments.data() );
 
 		colors[ 0 ] = red * 255;
```

```diff
diff --git a/src/cgame/cg_marks.cpp b/src/cgame/cg_marks.cpp
index dba1eaef0..10eca5630 100644
--- a/src/cgame/cg_marks.cpp
+++ b/src/cgame/cg_marks.cpp
@@ -183,7 +183,7 @@ void CG_ProcessImpactMarks()
 		byte           colors[ 4 ];
 		int            numFragments;
 		std::array<markFragment_t, MAX_MARK_FRAGMENTS> markFragments;
-		std::array<vec3_t, MAX_MARK_POINTS> markPoints;
+		std::array<std::array<float, 3>, MAX_MARK_POINTS> markPoints;
 		vec3_t         projection;
 
 		//if ( markTotal >= MAX_MARK_POLYS ) {
@@ -210,7 +210,7 @@ void CG_ProcessImpactMarks()
 		// get the fragments
 		VectorScale( dir, -20, projection );
 		numFragments = trap_CM_MarkFragments( 4, ( const vec3_t * ) originalPoints,
-			projection, MAX_MARK_POINTS, markPoints.data()[ 0 ],
+			projection, MAX_MARK_POINTS, markPoints.data()[ 0 ].data(),
 			MAX_MARK_FRAGMENTS, markFragments.data() );
 
 		colors[ 0 ] = red * 255;
```

(I modified the spacing in the diff so it fits in comment width)

Reply:
Done.